### PR TITLE
Add `query` option

### PR DESCRIPTION
**Pull request by arniu**
See [Help #46](https://github.com/fastify/help/issues/46)

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -1,19 +1,51 @@
 'use strict'
 
 const { Readable } = require('readable-stream')
-const Url = require('url')
+const qs = require('querystring')
 const util = require('util')
+const url = require('url')
+
+function parseURL (options) {
+  const parsedURL = url.parse(
```
Comment (delvedor) [inline lib/request.js]: As far as I know, `url.parse` is deprecated, you should use WHATWG URL API instead :)

Comment (mcollina) [reply lib/request.js]: You should use `const { URL} = require('url')` and then `new URL()`.

Comment (arniu) [reply lib/request.js]: The new WHATWG `URL` changes a lot. 

For example, `'/example?q=1'` is an invalid URL now:

```
new URL('/example?q=1')
{ TypeError [ERR_INVALID_URL]: Invalid URL: /example?q=1
    at onParseError (internal/url.js:240:17)
    at parse (internal/url.js:249:3)
    at new URL (internal/url.js:324:5) input: '/example?q=1' }
```

As a result, a `base` option may be required.

So, I thinks it's another task to mitigate to the new WHATWG `URL`.

Comment (mcollina) [reply lib/request.js]: I agree

**Comment by arniu** [comment]
@delvedor Would you please merge it?

**Comment by delvedor** [comment]
Hello! Sorry for the delay.
Do you want to work right away for the URL parsing migration? In that case, I'll wait before release a new version :)

**Pull request merged**